Bottom: f36ac99c2b81360c21d3a88ffbf8d6ee9369d66e
Top:    16d6e3339ece757bcc8d807072013e5f3692c6ec
Author: Tibor Palinkas <geda@igor2.repo.hu>
Date:   2010-12-22 21:04:17 +0000

hid/gtk: Use leaky_* functions for deliberate leaks in ghid_ui_info_append

Allocate cached temporary memory with the leak_* memory allocation wrapers
which allow these to be freed at exit.


---

diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index 90977e2..b1b75fa 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -3679,7 +3679,7 @@ ghid_ui_info_append (const gchar * new)
   if (new_ui_info_sz == 0) 
     {
       new_ui_info_sz = 1024;
-      new_ui_info = (gchar *) calloc ( new_ui_info_sz, sizeof (gchar));
+      new_ui_info = leaky_calloc (new_ui_info_sz, sizeof (gchar));
     }
 
   while (strlen (new_ui_info) + strlen (new) + 1 > new_ui_info_sz)
@@ -3688,7 +3688,7 @@ ghid_ui_info_append (const gchar * new)
       gchar * np;
 
       n = new_ui_info_sz + 1024;
-      if ( (np = realloc (new_ui_info, n)) == NULL)
+      if ((np = leaky_realloc (new_ui_info, n)) == NULL)
 	{
 	  fprintf (stderr, "ghid_ui_info_append():  realloc of size %ld failed\n",
 		   (long int) n);
